lib/Serialization/SerializedModuleLoader.cpp

```diff
@@ -81,12 +81,46 @@ openModuleFiles(StringRef DirName, StringRef ModuleFilename,
   return std::error_code();
 }
 
+static void addDiagnosticInfoForArchitectureMismatch(ASTContext &ctx,
+                             SourceLoc sourceLocation,
+                             llvm::SmallString<64> moduleName,
+                             llvm::SmallString<16> archName,
+                             llvm::SmallString<128> directoryPath) {
```

Comment:
Style notes:
- For parameters that would break the 80-column rule, please double-indent them from the left, rather than right-aligning them. (You can also use [`clang-format`](https://clang.llvm.org/docs/ClangFormat.html) on the code you changed to match our style if you want to set that up.)
- Since you aren't planning to modify any of these strings, the StringRef type is probably a better choice. There's an implicit conversion from SmallString and then the strings wouldn't get copied when you call the function.